src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -88591,9 +88594,50 @@ static void F(Pair<object?, object>? p)
     }
 }";
             var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
-            // https://github.com/dotnet/roslyn/issues/33006: Check nullability of `this`
-            // for generic `Deconstruct` method and infer nullability of type arguments.
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (12,34): warning CS8604: Possible null reference argument for parameter 'p' in 'void E.Deconstruct<object, object>(Pair<object, object> p, out object t, out object u)'.
```

Comment:
I understand why this is the warning we give, but this warning does not make much sense at first glance. I think it'd probably be worth introducing a custom message for extension deconstruction methods.

Reply:
I switched it to be 8602. Turns out it made the code simpler, and aligned the extension method warnings with the instance method ones. Arguably we could still have a specific warning for these cases, but I'm not sure if it adds much.

---
In reply to: [276820010](https://github.com/dotnet/roslyn/pull/35016#discussion_r276820010) [](ancestors = 276820010)

Reply:
Opened #35349 to track

---
In reply to: [278754133](https://github.com/dotnet/roslyn/pull/35016#discussion_r278754133) [](ancestors = 278754133,276820010)